packages/mui-system/src/colorManipulator/colorManipulator.test.js

```diff
@@ -303,126 +304,134 @@ describe('utils/colorManipulator', () => {
 
   describe('darken', () => {
     it("doesn't modify rgb black", () => {
-      expect(darken('rgb(0, 0, 0)', 0.1)).to.equal('rgb(0, 0, 0)');
+      expect(darken('rgb(0, 0, 0)', 0.1)).to.equal(hex('rgb(0, 0, 0)'));
     });
 
     it("doesn't overshoot if an above-range coefficient is supplied", () => {
       expect(() => {
-        expect(darken('rgb(0, 127, 255)', 1.5)).to.equal('rgb(0, 0, 0)');
+        expect(darken('rgb(0, 127, 255)', 1.5)).to.equal(hex('rgb(0, 0, 0)'));
       }).toErrorDev('MUI: The value provided 1.5 is out of range [0, 1].');
     });
 
     it("doesn't overshoot if a below-range coefficient is supplied", () => {
       expect(() => {
-        expect(darken('rgb(0, 127, 255)', -0.1)).to.equal('rgb(0, 127, 255)');
+        expect(darken('rgb(0, 127, 255)', -0.1)).to.equal(hex('rgb(0, 127, 255)'));
       }).toErrorDev('MUI: The value provided -0.1 is out of range [0, 1].');
     });
 
     it('darkens rgb white to black when coefficient is 1', () => {
-      expect(darken('rgb(255, 255, 255)', 1)).to.equal('rgb(0, 0, 0)');
+      expect(darken('rgb(255, 255, 255)', 1)).to.equal(hex('rgb(0, 0, 0)'));
     });
 
     it('retains the alpha value in an rgba color', () => {
-      expect(darken('rgb(0, 0, 0, 0.5)', 0.1)).to.equal('rgb(0, 0, 0, 0.5)');
+      expect(darken('rgb(0, 0, 0, 0.5)', 0.1)).to.equal(hex('rgb(0, 0, 0, 0.5)'));
     });
 
     it('darkens rgb white by 10% when coefficient is 0.1', () => {
-      expect(darken('rgb(255, 255, 255)', 0.1)).to.equal('rgb(229, 229, 229)');
+      expect(darken('rgb(255, 255, 255)', 0.1)).to.equal(hex('rgb(229, 229, 229)'));
     });
 
     it('darkens rgb red by 50% when coefficient is 0.5', () => {
-      expect(darken('rgb(255, 0, 0)', 0.5)).to.equal('rgb(127, 0, 0)');
+      expect(darken('rgb(255, 0, 0)', 0.5)).to.equal(hex('rgb(127, 0, 0)'));
     });
 
     it('darkens rgb grey by 50% when coefficient is 0.5', () => {
-      expect(darken('rgb(127, 127, 127)', 0.5)).to.equal('rgb(63, 63, 63)');
+      expect(darken('rgb(127, 127, 127)', 0.5)).to.equal(hex('rgb(63, 63, 63)'));
     });
 
     it("doesn't modify rgb colors when coefficient is 0", () => {
-      expect(darken('rgb(255, 255, 255)', 0)).to.equal('rgb(255, 255, 255)');
+      expect(darken('rgb(255, 255, 255)', 0)).to.equal(hex('rgb(255, 255, 255)'));
     });
 
     it('darkens hsl red by 50% when coefficient is 0.5', () => {
-      expect(darken('hsl(0, 100%, 50%)', 0.5)).to.equal('hsl(0, 100%, 25%)');
+      // Losing precision over 1/256th when rounding
+      // so we get #7f0000 instead of #800000
+      expect(darken('hsl(0, 100%, 50%)', 0.5)).to.equal(hex('#7f0000ff'));
```

Comment:
This is the worst case of rounding I could find, but I've also found it impossible to notice [a difference](https://jsfiddle.net/Luh2d4fy/2/).